File: Easy/2418. Sort the People.java
class Solution {
    public String[] sortPeople(String[] names, int[] heights) {
        Map<Integer, String> heightToName = new HashMap<>();
        int n = names.length;
        for (int i = 0; i < n; i++) {
            heightToName.put(heights[i], names[i]);
        }
        Arrays.sort(heights);
        for (int i = 0; i < n; i++) {
            names[i] = heightToName.get(heights[n - i - 1]);
        }
        return names;
    }
}

File: Easy/2418. Sort the People.kt
class Solution {
    fun sortPeople(names: Array<String>, heights: IntArray): Array<String> {
        val map = TreeMap<Int, String>()
        names.forEachIndexed {
            index, name -> map.put(heights[index], name)
        }
        return map.values.reversed().toTypedArray()
    }
}